module_fcst_grid_comp.F90

```diff
@@ -0,0 +1,968 @@
+
+           ! Add the field to the importState so parent can connect to it
+           call ESMF_StateAdd(exportState, (/fieldbundle/), rc=rc)
+           if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+         else if( i==2 ) then
+! for phys
+           nbdlphys = 2
+           allocate(fieldbundlephys(nbdlphys))
+           do j=1, nbdlphys
+             if( j==1 ) then
+               name_FB1 = trim(name_FB)//'_nearest_stod'
+             else
+               name_FB1 = trim(name_FB)//'_bilinear'
+             endif
+             fieldbundlephys(j) = ESMF_FieldBundleCreate(name=trim(name_FB1),rc=rc)
+             if (mype == 0) write(0,*)'af create fcst fieldbundle, name=',trim(name_FB1),'rc=',rc
+             if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+           enddo
+!
+           call fv_phys_bundle_setup(atm_int_state%Atm%diag, atm_int_state%Atm%axes, &
+                                     fieldbundlephys, fcstGrid, quilting, nbdlphys)
+!
+           ! Add the field to the importState so parent can connect to it
+           do j=1,nbdlphys
+             call ESMF_StateAdd(exportState, (/fieldbundlephys(j)/), rc=rc)
+             if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+           enddo
+
+         else
+
+           write(0,*)' unknown name_FB ', trim(name_FB)
+           ESMF_ERR_ABORT(101)
+
+         endif
+!
+        enddo
+
+!end qulting
+      endif
+
+      call get_atmos_model_ungridded_dim(nlev=numLevels, nsoillev=numSoilLayers,             &
+                                         ntracers=numTracers,                                &
+                                         num_diag_burn_emis_flux=num_diag_burn_emis_flux,    &
+                                         num_diag_sfc_emis_flux=num_diag_sfc_emis_flux,      &
+                                         num_diag_down_flux=num_diag_down_flux,              &
+                                         num_diag_type_down_flux=num_diag_type_down_flux,    &
+                                         num_diag_cmass=num_diag_cmass)
+!
+!-----------------------------------------------------------------------
+!
+      IF(rc /= ESMF_SUCCESS) THEN
+        WRITE(0,*)"FAIL: Fcst_Initialize."
+!      ELSE
+!        WRITE(0,*)"PASS: Fcst_Initialize."
+      ENDIF
+!
+      if (mype == 0) write(0,*)'in fcst,init total time: ', mpi_wtime() - timeis
+!
+!-----------------------------------------------------------------------
+!
+   end subroutine fcst_initialize
+!
+!-----------------------------------------------------------------------
+!#######################################################################
+!-----------------------------------------------------------------------
+!
+   subroutine fcst_run_phase_1(fcst_comp, importState, exportState,clock,rc)
```

Comment:
Need documentation or comments on what constitutes phase1 and phase2, etc. of the forecast grid comp.